src/main/java/de/oliver/fancyholograms/commands/hologram/SearchCMD.java

```diff
@@ -0,0 +1,75 @@
+package de.oliver.fancyholograms.commands.hologram;
+
+import com.google.common.primitives.Ints;
+import de.oliver.fancyholograms.FancyHolograms;
+import de.oliver.fancyholograms.api.Hologram;
+import de.oliver.fancyholograms.commands.Subcommand;
+import de.oliver.fancyholograms.util.Constants;
+import de.oliver.fancylib.MessageHelper;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class SearchCMD implements Subcommand {
+
+    @Override
+    public List<String> tabcompletion(@NotNull Player player, @Nullable Hologram hologram, @NotNull String[] args) {
+        return null;
+    }
+
+    @Override
+    public boolean run(@NotNull Player player, @Nullable Hologram hologram, @NotNull String[] args) {
+        final var holograms = FancyHolograms.get().getHologramsManager().getHolograms();
+
+        if (holograms.isEmpty()) {
+            MessageHelper.warning(player, "There are no holograms. Use '/hologram create' to create one");
+        } else {
+            int page;
+            String search = args[1];
+            if (args.length < 3) {
+                page = 1;
+            } else {
+                final var index = Ints.tryParse(args[2]);
+                if (index == null) {
+                    MessageHelper.error(player, "Could not parse page number");
+                    return false;
+                }
+                page = index;
+            }
+
+            var pages = holograms.stream()
```

Comment:
Instead of running 2 streams on holograms it could be worth moving the stream from line 52 here and creating an unmodifiable list with `toList()`. To check if the page exists you can then just run `List#isEmpty`.